restructuredtext_lint/cli.py

```diff
@@ -31,32 +47,39 @@ def _main(filepaths, format='text', stream=sys.stdout, encoding=None):
                     stream.write('{err.type} {err.source}:{err.line} {err.message}\n'.format(err=err))
             elif format == 'json':
                 error_dicts.extend({
-                    'line': error.line,
-                    'source': error.source,
-                    'level': error.level,
-                    'type': error.type,
-                    'message': error.message,
-                    'full_message': error.full_message,
-                } for error in file_errors)
+                    'line': err.line,
+                    'source': err.source,
+                    'level': err.level,
+                    'type': err.type,
+                    'message': err.message,
+                    'full_message': err.full_message,
+                } for err in file_errors)
 
     if format == 'json':
         stream.write(json.dumps(error_dicts))
 
     if error_occurred:
-        sys.exit(1)
+        sys.exit(2)  # Using 2 for linting failure, 1 for internal error
     else:
-        sys.exit(0)
+        sys.exit(0)  # Success!
 
 
 def main():
     # Set up options and parse arguments
-    parser = argparse.ArgumentParser(description='Lint reStructuredText files')
+    parser = argparse.ArgumentParser(description='Lint reStructuredText files. Returns 0 if all files pass linting, '
+                                     '1 for an internal error, and 2 if linting failed.')
     parser.add_argument('--version', action='version', version=VERSION)
     parser.add_argument('filepaths', metavar='filepath', nargs='+', type=str, help='File to lint')
     parser.add_argument('--format', default='text', type=str, help='Format of the output (e.g. text, json)')
     parser.add_argument('--encoding', type=str, help='Encoding of the input file (e.g. utf-8)')
+    parser.add_argument('--level', default='warning', type=str, choices=LEVEL_MAP,
+                        help='Minimum docutils linting error level to report and consider as failing '
+                        '(lower case string, default is warning)')
     args = parser.parse_args()
 
+    # Want the level strings to appear in the --help text via choices, so convert now:
+    args.level = LEVEL_MAP[args.level]
```

Comment:
`argparse` supports automatic resolution inside the `add_argument` itself. This was demonstrated in my original example:

https://docs.python.org/3/library/argparse.html#type